internal/validations/observability/common/validate.go

```diff
@@ -3,42 +3,58 @@ package common
-	if len(configs) == 0 {
-		return messages
-	}
-
 	for _, entry := range configs {
-		if entry.SecretName != "" {
-			if secret, found := secrets[entry.SecretName]; found {
-				if value, found := secret.Data[entry.Key]; !found {
-					messages = append(messages, fmt.Sprintf("secret[%s.%s] not found", entry.SecretName, entry.Key))
-				} else {
-					if len(value) == 0 {
-						messages = append(messages, fmt.Sprintf("secret[%s.%s] value is empty", entry.SecretName, entry.Key))
-					}
-				}
-			} else {
-				messages = append(messages, fmt.Sprintf("secret[%s] not found", entry.SecretName))
-			}
-		} else if entry.ConfigMapName != "" {
-			if cm, found := configMaps[entry.ConfigMapName]; found {
-				if value, found := cm.Data[entry.Key]; !found {
-					messages = append(messages, fmt.Sprintf("configmap[%s.%s] not found", entry.ConfigMapName, entry.Key))
-				} else {
-					if strings.TrimSpace(value) == "" {
-						messages = append(messages, fmt.Sprintf("configmap[%s.%s] value is empty", entry.ConfigMapName, entry.Key))
-					}
-				}
-			} else {
-				messages = append(messages, fmt.Sprintf("configmap[%s] not found", entry.ConfigMapName))
-			}
+		switch {
+		case entry.SecretName != "":
+			messages = append(messages, validateSecret(entry.SecretName, entry.Key, secrets)...)
+		case entry.ConfigMapName != "":
+			messages = append(messages, validateConfigMap(entry.ConfigMapName, entry.Key, configMaps)...)
 		}
 	}
+	return messages
+}
 
+func validateSecret(secretName, key string, secrets map[string]*corev1.Secret) (messages []string) {
+	secret, found := secrets[secretName]
+	if !found {
+		return []string{fmt.Sprintf("secret[%s] not found", secretName)}
+	}
+	if value, keyFound := secret.Data[key]; !keyFound {
+		messages = append(messages, fmt.Sprintf("secret[%s.%s] not found", secretName, key))
+	} else if len(value) == 0 {
+		messages = append(messages, fmt.Sprintf("secret[%s.%s] value is empty", secretName, key))
+	}
 	return messages
 }
+
+func validateConfigMap(configMapName, key string, configMaps map[string]*corev1.ConfigMap) (messages []string) {
+	cm, found := configMaps[configMapName]
+	if !found {
+		return []string{fmt.Sprintf("configmap[%s] not found", configMapName)}
+	}
+	if value, keyFound := cm.Data[key]; !keyFound {
+		messages = append(messages, fmt.Sprintf("configmap[%s.%s] not found", configMapName, key))
+	} else if strings.TrimSpace(value) == "" {
+		messages = append(messages, fmt.Sprintf("configmap[%s.%s] value is empty", configMapName, key))
+	}
+	return messages
+}
+
+// IsEnabledAnnotation checks if an annotation is set to either "true" or "enabled"
+func IsEnabledAnnotation(context internalcontext.ForwarderContext, annotation string) bool {
+	enabledValues := sets.NewString("true", "enabled")
+	if value, ok := context.Forwarder.Annotations[annotation]; ok {
+		if enabledValues.Has(value) {
```

Comment:
Do we care about case?

Reply:
We do.  Isn't everything else in the spec case sensitive?    Kubernetes standard is lowercase for annotations and labels, although I don't think they enforce it?    Also follows yaml standards afaik.